lib/services/waterline.js

```diff
@@ -125,8 +129,23 @@ function WaterlineServiceFactory(
         .then(function() {
             //it's OK to create indexes multiple times, so the code isn't protected by initialized
             //checking.
+            //
+            //For MongoDB 3.4+, creating indexes with same keys but different options throws errors.
+            //And the options `primaryKey: true` or `unique: true` in waterline models
+            //create indexes with option `{unique: true, sparse: true}` when waterline
+            //is initialized, so dropIndexes is called before createIndexes.
+            //
+            //Catching errors is because:
+            //  (1) When it fails to create indexes, the service still works.
+            //  (2) If errors are thrown, waterline isn't teardowned.
             return Promise.map(_.values(self.ontology.collections), function (collection) {
-                return collection.createIndexes();
+                return collection.dropIndexes()
```

Comment:
I think this is a dangerous action, You must be very careful to drop the whole existing indexes, because user my create their own indexes.

Is there any possibility to disable the waterline's index creation?